http-client/Network/HTTP/Client/Connection.hs

```diff
@@ -31,32 +31,31 @@ import Data.Function (fix)
 import Data.Maybe (listToMaybe)
 import Data.Word (Word8)
 
-connectionReadLine :: Maybe MaxHeaderLength -> Connection -> IO ByteString
+connectionReadLine :: MaxHeaderLength -> Connection -> IO ByteString
 connectionReadLine mhl conn = do
     bs <- connectionRead conn
     when (S.null bs) $ throwHttp IncompleteHeaders
     connectionReadLineWith mhl conn bs
 
 -- | Keep dropping input until a blank line is found.
-connectionDropTillBlankLine :: Maybe MaxHeaderLength -> Connection -> IO ()
+connectionDropTillBlankLine :: MaxHeaderLength -> Connection -> IO ()
 connectionDropTillBlankLine mhl conn = fix $ \loop -> do
     bs <- connectionReadLine mhl conn
     unless (S.null bs) loop
 
-connectionReadLineWith :: Maybe MaxHeaderLength -> Connection -> ByteString -> IO ByteString
+connectionReadLineWith :: MaxHeaderLength -> Connection -> ByteString -> IO ByteString
 connectionReadLineWith mhl conn bs0 =
     go bs0 id 0
   where
     go bs front total =
         case S.break (== charLF) bs of
             (_, "") -> do
-                let total' = total + S.length bs
-                case fmap unMaxHeaderLength mhl of
-                    Nothing -> pure ()
-                    Just n -> when (total' > n) $ throwHttp OverlongHeaders
+                when (total >= unMaxHeaderLength mhl && total /= 0) $ do
```

Comment:
This needs to compare against the new total, not the old one. And it should be checking to see if the `MaxHeaderLength` is 0, not the total. 
```suggestion
                let n = unMaxHeaderLength mhl
                    total' = total + fromIntegral (S.length bs)
                when (total' >= n && n /= 0) $ do
```

Reply:
I don't know why I made this change :)

Reply:
https://github.com/snoyberg/http-client/pull/549#discussion_r1890231380